### ISSUE TITLE
Arduino Uno

### ISSUE THREAD

**Issue by thijsforel** [RobotDynOfficial/Lib-RBD-Dimmer-for-Mega-UNO-Leonardo#4]
Hi,

Trying to use the library for the first time. Trying the SimpleDimmer example (provided with the Library) Arduino IDE says: 
"Global variables use 2337 bytes (114%) of dynamic memory, leaving -289 bytes for local variables. Maximum is 2048 bytes." 
 What's up with that?

**Comment by VSneg** [comment]
Hello, please try to rename the library folder on "RBDdimmer" and restart your Arduino IDE !

**Comment by rammnco** [comment]
Hi, i have the same issue. 
It seems that the RBDdimmer library alone is too much for the 2KB SRAM of the arduino uno. It uses too many global variables.
With the arduino mega as selected board, which has 8KB it compiles just fine.

Is this library compatible with Uno and Nano boards, as described in the README?

**Comment by pcex** [comment]
this works for atmega 328 it only uses arduino library but i need help aswell i want to keep the dim constant where this code fluctuates 
`/*
Pin    |  Interrrupt # | Arduino Platform
---------------------------------------
2      |  0            |  All -But it is INT1 on the Leonardo
3      |  1            |  All -But it is INT0 on the Leonardo
18     |  5            |  Arduino Mega Only
19     |  4            |  Arduino Mega Only
20     |  3            |  Arduino Mega Only
21     |  2            |  Arduino Mega Only
0      |  0            |  Leonardo
1      |  3            |  Leonardo
7      |  4            |  Leonardo
The Arduino Due has no standard interrupt pins as an iterrupt can be attached to almosty any pin. 
 
In the program pin 2 is chosen
*/
int AC_LOAD = 3;    // Output to Opto Triac pin
int dimming = 1024;  // Dimming level (0-128)  0 = ON, 128 = OFF
 
void setup()
{
  Serial.begin(115200);
  pinMode(AC_LOAD, OUTPUT);// Set AC Load pin as output
  attachInterrupt(0, zero_crosss_int, RISING);  // Choose the zero cross interrupt # from the table above
}
 
//the interrupt function must take no parameters and return nothing
void zero_crosss_int()  //function to be fired at the zero crossing to dim the light
{
  // Firing angle calculation : 1 full 50Hz wave =1/50=20ms 
  // Every zerocrossing thus: (50Hz)-> 10ms (1/2 Cycle) 
  // For 60Hz => 8.33ms (10.000/120)
  // 10ms=10000us
  // (10000us - 10us) / 128 = 75 (Approx) For 60Hz =>65
 
  int dimtime = (65*dimming);    // For 60Hz =>65    
  delayMicroseconds(dimtime);    // Wait till firing the TRIAC    
  digitalWrite(AC_LOAD, HIGH);   // Fire the TRIAC
  delayMicroseconds(8.33);         // triac On propogation delay 
         // (for 60Hz use 8.33) Some Triacs need a longer period
  digitalWrite(AC_LOAD, LOW);    // No longer trigger the TRIAC (the next zero crossing will swith it off) TRIAC
}
 
void loop()  {
for (int i=880; i <= 902; i++){
  if (i==902){
    dimming=0;
    delay(100);
    }
  else{
    dimming=i;
    Serial.println(i);
    delay(5);
  }
   }
 
for (int i=902; i >= 880; i--){
    if (i==902){
    dimming=0;
    delay(100);
    }
    else{
    dimming=i;
    Serial.println(i);
    delay(5);
    }
   }
   
}

`

**Comment by Gauthieer** [comment]
Hello !
I have the same issue as mentioned by @thijsforel with my Arduino Uno (with the SimplePotentiometer example) and your "AC Light Dimmer Module".

"Arduino IDE says: Global variables use 2411 bytes (117%) of dynamic memory, leaving -363 bytes for local variables. Maximum is 2048 bytes." So, impossible to compile the sketch.

As asked @rammnco, is this library compatible with Uno and Nano boards, as described in the README ? 

Thanks for your help,
Bests.

**Comment by JimmyBondi** [comment]
Same here...
I also would like to use this with an ATMega328 based Arduino.

**Comment by VSneg** [comment]
@rammnco @Gauthieer @JimmyBondi 
Did you try to rename the library folder on "RBDdimmer"?
Arduino\libraries\RBDdimmer

**Comment by rammnco** [comment]
@VSneg 
Yes, I have the folder named that way ("RBDdimmer"). 
I have tried it with the c and header files directly under this directory too.

Can this really be related to the library name though? The message says its about the global variables leaving no space for local variables...

It seems to be loaded well , because the sketch compiles just fine if I select a board with a bigger SRAM (like Arduino MEGA).
Just the Uno does'nt work, due to it's limited space. Same error mesasge like the others above.

**Comment by Gauthieer** [comment]
@VSneg
Yes, like @rammnco, my library folder's name is "RBDdimmer"... I also think it's just because the sketch is to big for Arduino Uno...Have you got a solution for that ?
Thanks.

**Comment by IvanKryukovRBD** [comment]
@JimmyBondi, @Gauthieer, @rammnco, @Gauthieer, @pcex, @thijsforel
We made new libarary please can you try and give us feedback )
https://github.com/RobotDynOfficial/RBDDimmer

**Comment by matt-mhd** [comment]
Has anyone resolved the memory issue with an Uno? trying to build a dimmer with potentiometers and get the same message. Have reloaded the RDBDimmer library and restarted Arduino IDE.... Is it that I just need to use a Mega with bigger memory?

**Comment by Garithes** [comment]
Auch mit der neuen Version:
Globale Variablen verwenden 2139 Bytes (104%) des dynamischen Speichers, -91 Bytes für lokale Variablen verbleiben. Das Maximum sind 2048 Bytes.
Nicht genug Arbeitsspeicher; unter http://www.arduino.cc/en/Guide/Troubleshooting#size finden sich Hinweise, um die Größe zu verringern.
Fehler beim Kompilieren für das Board Arduino/Genuino Uno.

**Comment by GabrielLAPU** [comment]
Hi, I try to connect an Arduino Nano Every but the library don't work with this version of Arduino, anypeople have idea how I can operate with this library in this Arduino Model?
Thanks in advance. 
Regards.
